connector/saml/saml.go

```diff
@@ -334,19 +353,97 @@ func (p *provider) HandlePOST(s connector.Scopes, samlResponse string) (ident co
 	return ident, nil
 }
 
+func (p *provider) validateStatus(resp *response) error {
+	status := resp.Status
+	if status == nil {
+		return fmt.Errorf("response did not contain a Status")
+	}
+	statusCode := status.StatusCode
+	if statusCode == nil {
+		return fmt.Errorf("response did not contain a StatusCode")
+	}
+	if statusCode.Value != "urn:oasis:names:tc:SAML:2.0:status:Success" {
```

Comment:
I think it make sense to return the `StatusMessage` if present to the end user. I will add this element to the status struct. If not present a human reable message containing the statusCode should be returned. This way it is done in the [python library](https://github.com/onelogin/python-saml/blob/master/src/onelogin/saml2/response.py#L317-L335). 

As these elements are optional in the response of course it should not give an error if they are not found. I will change it accordingly.